Editor/EditorCore/EditorSceneViewPicker.cs

```diff
@@ -520,7 +520,13 @@ static float VertexRaycast(Vector3 mousePosition, ScenePickerPreferences pickerO
             Camera cam = SceneView.lastActiveSceneView.camera;
             selection.Clear();
             s_NearestVertices.Clear();
-            selection.gameObject = HandleUtility.PickGameObject(mousePosition, false);
+            bool highlightedVertexExists = ProBuilderEditor.instance.hovering.vertexes.Count > 0;
+
+            // if there is a vertex already highlighted, we don't want to select a different game object
+            if (!highlightedVertexExists)
+            {
+                selection.gameObject = HandleUtility.PickGameObject(mousePosition, false);
+            }
```

Comment:
It would be better to refactor this duplicated code into a helper method. It would ensure that future changes only need to be made in a single location as whoever might be touching it in future may not be aware it's duplicated in EdgeRaycast.

Reply:
@katherinesheridan

Reply:
OOh
You mean :
```
void PickObjectIfNotSubElementSelected(Vector2 mousePosition, bool highlightElementExists)
{
            if (!highlightedElementExists)
                selection.gameObject = HandleUtility.PickGameObject(mousePosition, false);
}
```

With basically: 
`PickObjectIfNoSubElementSelected(mousePosition. ProBuilderEditor.instance.hovering.vertexes.Count > 0);`
 ?

Reply:
```
void PickIfElementNotHighlighted(Vector2 mousePosition, ComponentMode elementType)
        {
            bool elementHighlighed = false;

            switch (elementType)
            {
                case ComponentMode.Vertex:
                    elementHighlighed = ProBuilderEditor.instance.hovering.vertexes.Count > 0;
                    break;
                
                case ComponentMode.Edge:
                    elementHighlighed = ProBuilderEditor.instance.hovering.edges.Count > 0;
                    break;
            }

            // Pick only if no element is highlighted
            if (!elementHighlighed)
            {
                selection.gameObject = HandleUtility.PickGameObject(mousePosition, false);
            }
        }
    }
```
then for verts do:
`PickIfElementNotHighlighted(mousePos, ComponentType.vertex)`

then for edges do:
`PickIfElementNotHighlighted(mousePos, ComponentType.edges)`

Reply:
Anyhow, more a recommendation rather than a blocker - I'll leave it to Katherine to decide. The fix itself is great 👍